syntax = "proto3";

package user;

service UserService {
  rpc Register (RegisterRequest) returns (RegisterResponse);
  rpc Auth (AuthRequest) returns (AuthResponse);
  rpc GetUserInfo (GetUserInfoRequest) returns (GetUserInfoResponse);
}

message RegisterRequest {
  string username = 1;
  string password = 2;
  string role = 3;
}

message RegisterResponse {
  bool success = 1;
  bool response_success = 2;
}

message AuthRequest {
  string username = 1;
  string password = 2;
}

message AuthResponse {
  bool success = 1;
  bool response_success = 2;
}

message GetUserInfoRequest {
  string username = 1;
}

message GetUserInfoResponse {
  string username = 1;
  string role = 2;
}
